### PR TITLE
Use ChunkMetaActor to replace KVStoreActor

### PR DESCRIPTION
**Pull request by wjsi**
Use ChunkMetaActor to replace KVStoreActor as KVStoreActor is single-pointed and introduces the cost of path formatting and resolving. This resolves https://github.com/mars-project/mars/issues/66.

### CODE REVIEW

File: mars/scheduler/graph.py [hunk 1]
```diff
@@ -534,24 +600,25 @@ def get_executable_operand_dag(self, op_key, serialize=True):
         else:
             return graph
 
-    def create_operand_actors(self):
+    def create_operand_actors(self, **kwargs):
```
Comment (chineking) [inline mars/scheduler/graph.py]: Better to check `kwargs` if there's any unexpected param.

File: mars/api.py [hunk 2]
```diff
@@ -70,16 +68,24 @@ def delete_graph(self, session_id, graph_key):
...
-        state = state_obj.value if state_obj else 'preparing'
+        from .scheduler import GraphState
+
+        graph_meta_uid = GraphMetaActor.gen_name(session_id, graph_key)
+        graph_meta_ref = self.get_actor_ref(graph_meta_uid)
+        if self.actor_client.has_actor(graph_meta_ref):
+            state_obj = graph_meta_ref.get_state()
+            state = state_obj.value if state_obj else 'preparing'
+        else:
+            state = 'preparing'
```
[the remark this answers is not in the document]
Comment (wjsi) [reply mars/api.py]: GraphMetaActor is created with GraphActor, thus it is not always available.

File: mars/scheduler/chunkmeta.py [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+
+    def get_worker_chunks(self, worker, default=None):
+        """
+        Get chunks held in a worker
+        :param worker: worker endpoint
+        :param default: default value
+        """
+        return self._worker_chunks.get(worker, default)
+
+    def remove_worker_keys(self, worker, filter_fun=None):
```
[the remark this answers is not in the document]
Comment (wjsi) [reply mars/scheduler/chunkmeta.py]: ChunkMetaStore is not an actor, thus there is no inter-process operations.

File: mars/scheduler/chunkmeta.py [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+        for ckey in tuple(self._worker_chunks[worker]):
+            if not filter_fun(ckey):
+                continue
+            self._del_worker_in_metas(ckey, (worker,))
+            meta = store[ckey]
+            new_workers = meta.workers = tuple(w for w in meta.workers if w != worker)
+            if not new_workers:
+                affected.append(ckey)
+        for ckey in affected:
+            del self[ckey]
```
[the remark this answers is not in the document]
Comment (wjsi) [reply mars/scheduler/chunkmeta.py]: ``_del_worker_in_metas`` (now ``_del_chunk_key_from_workers``) merely remove chunks in a worker, thus we need to remove the meta if no workers hold the corresponding data.

File: mars/scheduler/chunkmeta.py [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+        if not self._worker_chunks[worker]:
+            del self._worker_chunks[worker]
```
[the remark this answers is not in the document]
Comment (wjsi) [reply mars/scheduler/chunkmeta.py]: When the worker holds nothing, it should be removed.

File: mars/scheduler/graph.py [hunk 6]
```diff
@@ -82,6 +80,49 @@ def fetch_tensor(self, session_id, graph_key, tensor_key):
...
+
+    def __init__(self, session_id, graph_key):
+        super(GraphMetaActor, self).__init__()
+        self._session_id = session_id
+        self._graph_key = graph_key
+
+        self._kv_store_ref = None
+
+        self._state = None
+        self._final_state = None
```
[the remark this answers is not in the document]
Comment (wjsi) [reply mars/scheduler/graph.py]: First used in GraphActor, just a simple copy.

File: mars/scheduler/chunkmeta.py [hunk 7]
```diff
@@ -0,0 +1,437 @@
...
+            futures.append(self.ctx.actor_ref(LocalChunkMetaActor.default_name(), address=addr) \
+                           .batch_get_chunk_meta(session_id, keys, _wait=False))
+
+        # accept results and merge
+        for keys, future in zip(query_dict.values(), futures):
+            results = future.result()
+            meta_dict.update(zip(keys, results))
+        return [meta_dict.get(k) for k in chunk_keys]
+
+    def delete_meta(self, session_id, chunk_key):
```
[the remark this answers is not in the document]
Comment (wjsi) [reply mars/scheduler/chunkmeta.py]: Broadcasting to registered eps added in LocalChunkMetaActor.

**Pull request merged**